src/Simulation/Common/QubitManager.cs

```diff
@@ -259,7 +259,6 @@ public virtual Qubit Allocate()
         /// </summary>
         public virtual IQArray<Qubit> Allocate(long numToAllocate)
         {
-            IgnorableAssert.Assert(numToAllocate >= 0, "Attempt to allocate negative number of qubits.");
             if (numToAllocate < 0)
```

Comment:
I would strongly suggest against removing asserts like this. In fact, just revert changes to this file. The fix you did to the test file is proper and resolves all issues. As for this, I do not see any harm in the presence of assert, and leaving it in helps debugging (without adding any drawbacks).